sdks/go/pkg/beam/runners/prism/internal/worker/worker_test.go

```diff
@@ -87,19 +140,22 @@ func serveTestWorker(t *testing.T) (context.Context, *W, *grpc.ClientConn) {
 	ctx, cancelFn := context.WithCancel(context.Background())
 	t.Cleanup(cancelFn)
 
-	w := New("test", "testEnv")
-	lis := bufconn.Listen(2048)
-	w.lis = lis
```

Comment:
I will note that we used bufconn here to avoid actually using a network port during the unit tests. Not a hard blocker, but it would be nice to try to maintain that instead of using a new port every time.

What lead you to stop using the buffered connection?

Reply:
I was able to add it back. I saw a deprecated warning about the use of DialContext but I see you had it before causing no issues so I added it back.